libraries/type-length-value/src/state.rs

```diff
@@ -891,57 +889,60 @@ mod test {
             ProgramError::InvalidAccountData,
         );
     }
-}
-#[cfg(all(test, feature = "borsh"))]
-mod borsh_test {
-    use super::*;
-    #[derive(Clone, Debug, PartialEq, borsh::BorshDeserialize, borsh::BorshSerialize)]
-    struct TestBorsh {
+
+    #[derive(Clone, Debug, PartialEq)]
+    struct TestUnsized {
         data: String, // test with a variable length type
-        inner: TestInnerBorsh,
     }
-    #[derive(Clone, Debug, PartialEq, borsh::BorshDeserialize, borsh::BorshSerialize)]
-    struct TestInnerBorsh {
-        data: String,
-    }
-    impl SplDiscriminate for TestBorsh {
+    impl SplDiscriminate for TestUnsized {
         const SPL_DISCRIMINATOR: ArrayDiscriminator =
             ArrayDiscriminator::new([5; ArrayDiscriminator::LENGTH]);
     }
+    impl UnsizedPack for TestUnsized {
+        fn pack_into_slice(&self, dst: &mut [u8]) -> Result<(), ProgramError> {
+            dst[..8].copy_from_slice(&self.data.len().to_le_bytes());
+            dst[8..].copy_from_slice(self.data.as_bytes());
```

Comment:
`copy_from_slice` does require that src and dst have the same length, unfortunately

Reply:
```suggestion
            let bytes = self.data.as_bytes();    
            dst[8..8+bytes.len()].copy_from_slice(bytes);
```
Does this work?

Reply:
yep! I just needed to add a length check too to avoid the panic